mono/metadata/unity-memory-info.c

```diff
@@ -24,67 +24,84 @@ typedef struct CollectMetadataContext
+
+		fieldCount = mono_class_num_fields (klass);
+		
+		if (fieldCount > 0) {
+			while ((field = mono_class_get_fields (klass, &iter))) {
+				MonoClass *fieldKlass = mono_class_from_mono_type (field->type);
+
+				if (fieldKlass != klass)
+					ContextRecurseClassData (context, fieldKlass);
+			}
+		}
+	}
 }
 
-static void CollectHashMapClass(gpointer key, gpointer value, gpointer user_data)
+static void
+CollectHashMapClass (gpointer key, gpointer value, gpointer user_data)
 {
-	CollectMetadataContext* context = (CollectMetadataContext*)user_data;
-	MonoClass* klass = (MonoClass*)value;
-	ContextInsertClass(context, klass);
+	CollectMetadataContext *context = (CollectMetadataContext *)user_data;
+	MonoClass *klass = (MonoClass *)value;
+	ContextRecurseClassData (context, klass);
 }
 
-static void CollectHashMapListClasses(gpointer key, gpointer value, gpointer user_data)
+static void
+CollectHashMapListClasses (gpointer key, gpointer value, gpointer user_data)
 {
-	CollectMetadataContext* context = (CollectMetadataContext*)user_data;
-	GSList* list = (GSList*)value;
+	CollectMetadataContext *context = (CollectMetadataContext *)user_data;
+	GSList *list = (GSList *)value;
 
-	while (list != NULL)
-	{
-		MonoClass* klass = (MonoClass*)list->data;
-		ContextInsertClass(context, klass);
+	while (list != NULL) {
+		MonoClass *klass = (MonoClass *)list->data;
+		ContextRecurseClassData (context, klass);
 
-		list = g_slist_next(list);
+		list = g_slist_next (list);
 	}
 }
 
-static void CollectGenericClass(MonoGenericClass* genericClass, gpointer user_data)
+static void
+CollectGenericClass (MonoGenericClass *genericClass, gpointer user_data)
 {
-	CollectMetadataContext* context = (CollectMetadataContext*)user_data;
+	CollectMetadataContext *context = (CollectMetadataContext *)user_data;
 
 	if (genericClass->cached_class != NULL)
-		ContextInsertClass(context, genericClass->cached_class);
+		ContextRecurseClassData (context, genericClass->cached_class);
 }
 
-static void CollectImageMetaData(MonoImage* image, gpointer value, CollectMetadataContext* context)
+static void
+CollectImageMetaData (MonoImage *image, gpointer value, CollectMetadataContext *context)
 {
 	int i;
 	MonoTableInfo *tdef = &image->tables[MONO_TABLE_TYPEDEF];
 	GSList *list;
 
-	if (image->dynamic)
-	{
+	if (image->dynamic) {
 		GHashTableIter iter;
 		gpointer key;
-		MonoDynamicImage* dynamicImage = (MonoDynamicImage*)image;
-
-		g_hash_table_iter_init(&iter, dynamicImage->typeref);
+		MonoDynamicImage *dynamicImage = (MonoDynamicImage *)image;
+		g_hash_table_iter_init (&iter, dynamicImage->typeref);
 
-		while (g_hash_table_iter_next(&iter, &key, NULL))
-		{
-			MonoType* monoType = (MonoType*)key;
-			MonoClass* klass = mono_type_get_class(monoType);
+		while (g_hash_table_iter_next (&iter, &key, NULL)) {
+			MonoType *monoType = (MonoType *)key;
+			MonoClass *klass = mono_type_get_class (monoType);
```

Comment:
I might want to be using `mono_class_from_mono_type` here rather than `mono_type_get_class`

Reply:
alright will improve

Reply:
changed in 8b1b682